File: book-network-ui/src/app/models/user-profile-dto.ts
// Create this file at src/app/models/user-profile-dto.ts
export interface UserProfileDto {
  username: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  id?: string;
  profilePictureUrl?: string;
  booksCount?: number;
  sharedBooksCount?: number;
}

File: src/main/java/com/alibou/book/user/UserProfileDto.java
package com.alibou.book.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileDto {
    private String id;
    private String username;
    private String firstName;
    private String lastName;
    private String email;
    private String bio;
    private String profilePictureUrl;
    private int booksCount;
}